### ISSUE TITLE
Refactor the keratorch code 

### ISSUE THREAD

**Issue by JamorMoussa** [JamorMoussa/keratorch#1]
**Issue Description**

Need to make some refactoring of keratorch code: 

- `kt.nn.ktModule`: is too big, need to create `ktTrainer`, and make the `ktModule` extends from it.
- Add a `metrics` argument to the `fit` method: `model.fit(metrics = [kt.metics.Accuracy()])` .
- move the `LossCallBack` to metrics.
- move `Sate` to the main directory of `keratorch`.
- move the `Lambda` to `layer`.
